### DISCUSSION TITLE
Restored from v. 2024.08.8 to 2024.05.7

### DISCUSSION THREAD

**Discussion by wickzter** [wills106/homeassistant-solax-modbus#990] [Q&A]
I decided to upgrade the integration today, took a full backup first, deleted the inverter, restarted HA, upgraded the integration, restarted HA, added the inverter back.  My automations showed errors about unknown device, and I was experiencing the naming issue with the new prefix being added.  I could live with fixing the automations, but later this afternoon I discovered when I made changes to settings on my dashboard they weren't being replicated in Solax cloud.  I don't have enough time today to try and figure it all out so I reverted to my backup and it all seems good again.  This is a brilliant integration and thanks go to Wills106 for it, we'd be stuck without him.  And thanks to all you guys that are smarter than I am at coding etc and will no doubt have this figured out, you are all awesome!!

**Comment by mapotn** [comment]
2024.05.7 to 2024.08.8
I have also decided to upgrade the integration 2024.05.7 to 2024.08.8
so deleted the inverter, upgraded the integration, restarted HA, added the inverter back.
Surprisingly I am not able to connect via USB modbus adaptor anymore
![image](https://github.com/user-attachments/assets/f70eec53-9bae-4cc1-a608-48496f3e5bce)

but it looks like works only via TCP modbus (with inverter's IP address) which was not working until now with my X1 Boost Gen3
It would be fine but I have noticed that entity SolaX Inverter Inverter Voltage and SolaX Inverter Inverter Frequency
 is unknown from now:
![image](https://github.com/user-attachments/assets/56df029f-538b-4d52-a16e-247ffa1ed018)
![image](https://github.com/user-attachments/assets/4ceb46f6-4025-46c0-93a2-21db32dfa942)

@wills106 Do you have any advice please?

Anyway thanks for your perfect work!

**Comment by MHBraun** [comment]
It would be helpful if there is a description of clear path how to upgrade from 2024.05.07 to 2024.08.08 without any  issues.
Just deleting the integration, restarting and reinstalling seems not to work.
I will stay at 2024.05.07 as it is just working fine.

**Reply by michaelschwarz** [reply]
I am also stuck on 2024.05.2. Some weeks ago I updated, and as latest release notes didn't show the warnings any more I lost most of my stuff, and decided immediately to restore a backup I made before.

Yes, please, if we can have a clear step-by-step documentation would be nice. I am afraid to update (and more because of my wife will kill me if other stuff will not work).   ;-)
